src/server/network.ts

```diff
@@ -263,6 +264,24 @@ export type ResourceTiming = {
   responseStart: number;
 };
 
+export type IPAddressAndPort = {
+  ipAddress?: string;
+  port?: number;
+}
+
+export type SecurityDetails = {
+    protocol?: string;
```

Comment:
Are these all actually optional in the final HAR? I'd expect to always have a protocol.

Reply:
Yes, everything in the Security domain in WK is marked as optional: https://github.com/WebKit/WebKit/blob/8afe31a018b11741abdf9b4d5bb973d7c1d9ff05/Source/JavaScriptCore/inspector/protocol/Security.json

And this holds in practice. See branching in the tests which I arrived at watching bot fails.